pkg/cmd/get/klusterletinfo/exec.go

```diff
@@ -0,0 +1,245 @@
+// Copyright Contributors to the Open Cluster Management project
+package klusterletinfo
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/kubernetes"
+	operatorclient "open-cluster-management.io/api/client/operator/clientset/versioned"
+	v1 "open-cluster-management.io/api/operator/v1"
+	"open-cluster-management.io/clusteradm/pkg/helpers/printer"
+)
+
+func (o *Options) complete(cmd *cobra.Command, args []string) error {
+	cfg, err := o.ClusteradmFlags.KubectlFactory.ToRESTConfig()
+	if err != nil {
+		return err
+	}
+	operatorClient, err := operatorclient.NewForConfig(cfg)
+	if err != nil {
+		return err
+	}
+	kubeClient, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		return err
+	}
+	crdClient, err := clientset.NewForConfig(cfg)
+	if err != nil {
+		return err
+	}
+
+	o.kubeClient = kubeClient
+	o.operatorClient = operatorClient
+	o.crdClient = crdClient
+	return nil
+}
+
+func (o *Options) validate(args []string) error {
+	if len(args) != 0 {
+		return fmt.Errorf("there should be no argument")
+	}
+	return nil
+}
+
+const (
+	klusterletName                = "klusterlet"
+	registrationOperatorNamespace = "open-cluster-management"
+	klusterletCRD                 = "klusterlets.operator.open-cluster-management.io"
+
+	componentsNamespace            = "open-cluster-management-agent"
+	componentNameRegistrationAgent = "klusterlet-registration-agent"
+	componentNameWorkAgent         = "klusterlet-work-agent"
+)
+
+func (o *Options) run() error {
+	// printing registration-operator
```

Comment:
I think we should also consider showing the klusterlet API info here. There is some very helpful info like whether the klusterlet secret is correctly configured.

Reply:
Will we print the whole API info with the tree-style?

Reply:
Or we just choose some to show? such as Spec.DeployOption, Status.Conditions?

Reply:
I would think conditions at first. Actually since it has relatedObject and generations in the status. It probably should read resource from there first and then get the certain resources.